testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -323,17 +397,64 @@ def u():
+                                                      use_kdtree=True)()
+        vers2 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                      'protein', 2,
+                                                      not_within=False,
+                                                      use_kdtree=False)()
 
         assert_equal(vers1, vers2)
 
     def test_not_within(self, u):
-        from MDAnalysis.analysis.density import notwithin_coordinates_factory as ncf
-
-        vers1 = ncf(u, 'resname SOL', 'protein', 2, not_within=True, use_kdtree=True)()
-        vers2 = ncf(u, 'resname SOL', 'protein', 2, not_within=True, use_kdtree=False)()
-
+        vers1 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                     'protein', 2,
+                                                     not_within=True,
+                                                     use_kdtree=True)()
+        vers2 = density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                     'protein', 2,
+                                                     not_within=True,
+                                                     use_kdtree=False)()
         assert_equal(vers1, vers2)
+
+    def test_has_DeprecationWarning(self, u):
+        with pytest.warns(DeprecationWarning,
+                          match="will be removed in release 2.0.0"):
+            density.notwithin_coordinates_factory(u, 'resname SOL',
+                                                  'protein', 2,
+                                                  not_within=False,
+                                                  use_kdtree=True)()
+
+
+class TestBfactor2RMSF(object):
+    @pytest.mark.parametrize('B', [20, 20.0, 20.123,
+                                   np.array([45.678, 67.1, 0.0, 100.2])])
+    def test_Bfactor2RMSF(self, B):
+        values = density.Bfactor2RMSF(B)
+        ref = np.sqrt(3. * B / 8.) / np.pi  # original equation
+        assert_almost_equal(values, ref)
+
+    def test_has_DeprecationWarning(self):
+        with pytest.warns(DeprecationWarning,
+                          match="will be removed in release 2.0.0"):
+            density.Bfactor2RMSF(100.0)
+
+class Test_density_from_PDB(object):
+
+    @pytest.mark.parametrize('sigma,ref_shape,ref_gridsum',
+                             [(None, (21, 26, 29), 108.710149),
+                              (1.5,  (21, 26, 29), 160.050167)])
+    def test_density_from_PDB(self, sigma, ref_shape, ref_gridsum):
+        # simple test that the code runs: use B-factors from PDB or fixed sigma
+        D = density.density_from_PDB(PDB_full, delta=2.0, sigma=sigma)
+
+        assert isinstance(D, density.Density)
+        assert_equal(D.grid.shape, ref_shape)
+        assert_almost_equal(D.grid.sum(), ref_gridsum, decimal=6)
```

Comment:
Looks like the numpy 1.13.3 tests are failing here because it only matches to 5 d.p. instead of 6.

Reply:
need to decrease `decimal=5`